lib/python/rose/suite_control.py

```diff
@@ -71,9 +79,16 @@ def gcontrol(self, suite_name, host=None, confirm=None, stderr=None,
 
         """
         engine_version = self._get_engine_version(suite_name)
-        for host in self._get_hosts(suite_name, host):
+        if host:
+            host_names = [host]
+        else:
+            host_names = self.suite_engine_proc.get_suite_run_hosts(
+                None, suite_name)
+        if not host_names:
+            raise SuiteNotRunningError(suite_name)
+        for host_name in host_names:
             self.suite_engine_proc.gcontrol(
-                suite_name, host, engine_version, args)
+                suite_name, host_name, engine_version, args)
```

Comment:
For safety. In case users have somehow managed to start the suite on multiple hosts (which is possible, especially on sites with suite hosts not sharing a file system).